lib/compile_time.flow

```diff
@@ -0,0 +1,20 @@
+export {
+	// Speculative function, used to make output of computed constant values during compilation.
+	// Normally these function is eliminated from program before passing it to backends.
+	outputCt(msg: string, v: ?) -> ? {
```

Comment:
call this "compileTime".

Reply:
What is the difference between outputCompileTimeEval and forceCompiletimeEval?

Also, "output" makes no sense at all. "compileTime" makes a lot of sense, like I wrote. No need to say "eval" - that is not required.

Reply:
1) outputCompileTimeEval: _prints_ the computed expression on the screen during compilation. If computation of an expression fails (i.e. it's not a compile-time constant), the original expression is printed. Otherwise the obtained constant is printed as an expression. The main reason for this function: make it visible for a programmer, what is being computed during compile-time. I think it's a useful function, since I would like to have a simple way of knowing values of some compile-time constants

2) forceCompiletimeEval: the directive to the compile-time evaluation engine to raise an error, if the computation of a constant during compile-time fails. On my point, this function make sense when a programmer intentionally wants to perform some computations during compile-time and wants to be sure, that these computations really succeeded. 

About 'Eval' suffix - ok, may be easily dropped.

Reply:
Isn't outputCompiletime just "compileTime(println(whatever))"? Seems completely unnecessary.

"compileTime" for the other one seems like the most logical name.

Reply:
> Isn't outputCompiletime just "compileTime(println(whatever))"? Seems completely unnecessary.
> 
> "compileTime" for the other one seems like the most logical name.

Currently `compileTime(println(whatever))` won't work, because `println` is IO native (not pure), i.e. is not computed at runtime. Also `println` doesn't return value, thus can't be used as an intermediate layer, which prints a value and then passes it further. 

Another reason, why `compileTime` doesn't replace `outputCompileTime` is that currently the model of compile-time computations is implicit, i.e. the compiler tries to compute _any_ expression, and if succeeds, it may replace the expression with a constant. Thus, `compileTime` is implicitly put on _every_ expression, and in order to distinguish in which expression we are interested in during the compile time we need some markup function (which outputCompileTime exactly is).